go/apps/mixnet/mixnet_test.go

```diff
@@ -286,7 +270,13 @@ func TestProxyRouterRelay(t *testing.T) {
 		if res.err != nil {
 			t.Error(res.err)
 		} else if bytes.Compare(res.msg, msg[:l]) != 0 {
-			t.Error("relay (length=%d): Server got: %s", l, res.msg)
+			t.Errorf("relay (length=%d): Server got: %s", l, res.msg)
+		}
+
+		if bytes.Compare(reply, msg[:l]) != 0 {
+			t.Errorf("relay (length=%d): reply does not match message", l)
```

Comment:
I think it would be clearer to include reply and msg in the top Errorf and use %v and % x respectively to print them out.